File: C++/프로그래머스/2/131704. 택배상자/택배상자.cpp
#include <stack>
#include <vector>

using namespace std;

int solution(vector<int> order) {
    stack<int> st;
    st.push(0);
    int answer = 0, container = 1;
    
    for(auto& o : order) {
        while(st.top() < o) {
            st.push(container++);
        }
        
        if(st.top() != o) {
            break;
        }
        answer++;
        st.pop();
    }
    
    return answer;
}

File: Python3/프로그래머스/2/131704. 택배상자/택배상자.py
from collections import deque

def solution(order):
    stack = [0]
    answer = 0
    
    curCt = 1
    for o in order:
        while curCt < o:
            stack.append(curCt)
            curCt += 1
            
        if curCt == o:
            answer += 1
            curCt += 1
        elif stack[-1] == o:
            answer += 1
            del stack[-1]
        else:
            break
    
    return answer